arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcData.java

```diff
@@ -106,10 +106,53 @@ public void callEcData(
       case PRC_ECMUL -> ecMulEffectiveCall.addPrecompileLimit(1);
       case PRC_ECRECOVER -> ecRecoverEffectiveCall.addPrecompileLimit(1);
       case PRC_ECPAIRING -> {
-        // TODO: @Lorenzo @Olivier complete
-        //  ecPairingG2MembershipCalls.addPrecompileLimit();
-        //  ecPairingMillerLoops.addPrecompileLimit();
-        //  ecPairingFinalExponentiations.addPrecompileLimit();
+        // TODO: @Olivier @Lorenzo: review
+
+        // ecPairingG2MembershipCalls case
+        // NOTE: see EC_DATA specs Figure 3.5 for a graphical representation of this case analysis
+        if (!ecDataOperation.internalChecksPassed()) {
+          ecPairingG2MembershipCalls.addPrecompileLimit(0);
+          // The circuit is never invoked in the case of internal checks failing
+        }
```

Comment:
Is this serving a purpose or is it for clarification ? If it is for clarification shouldn't we just put 0 for the other two circuits, too ?